### PR TITLE
[OSDEV-981] Reporting. History of contributor uploads.

### PR DESCRIPTION
**Pull request by Innavin369**
[OSDEV-981] (https://opensupplyhub.atlassian.net/browse/OSDEV-981) Reporting. History of contributor uploads.

[OSDEV-981]: https://opensupplyhub.atlassian.net/browse/OSDEV-981?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: doc/release/RELEASE-NOTES.md [hunk 1]
```diff
@@ -28,6 +28,7 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 * [OSDEV-705](https://opensupplyhub.atlassian.net/browse/OSDEV-705) - Fixed the error “could not convert string to float” that occurred when a list contained columns for “lat” and “lng” and only some of the rows in these columns had data. As a result, rows are processed regardless of whether the values for “lat” and “lng” are present and valid, invalid, or empty.
 
 ### What's new
+* [OSDEV-981](https://opensupplyhub.atlassian.net/browse/OSDEV-981) Reporting. History of contributor uploads.
```
Comment (vladsha-dev) [inline doc/release/RELEASE-NOTES.md:31]: Could you please describe what was done? It looks like that is the name of the task in Jira.

File: src/django/api/reports/history_of_contributor_uploads.sql [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  COALESCE(CURRENT_DATE - min(l.created_at)::date,CURRENT_DATE - min(s.created_at)::date) AS "Time Since First Upload (Days)",
+  COALESCE(TO_CHAR(max(l.created_at), 'Month DD, YYYY'),TO_CHAR(max(s.created_at), 'Month DD, YYYY')) AS "Date of Last Upload",
+  COALESCE(CURRENT_DATE - max(l.created_at)::date,CURRENT_DATE - max(s.created_at)::date) AS "Time Since Last Upload (Days)",
+  COUNT(*) AS "Lifetime Uploads",
+  COALESCE(ROUND(COUNT(*) / ROUND((CURRENT_DATE - min(l.created_at)::date) / 365.25, 2), 2),ROUND(COUNT(*) / ROUND((CURRENT_DATE - min(s.created_at)::date) / 365.25, 2), 2))  AS "Uploads per Year"
+FROM api_contributor ac
+LEFT JOIN api_source s ON s.contributor_id = ac.id
+LEFT JOIN api_facilitylist l on l.id = s.facility_list_id
+LEFT JOIN api_user u ON ac.admin_id = u.id
+where s.create = true
```
Comment (vladsha-dev) [inline src/django/api/reports/history_of_contributor_uploads.sql:15]: Please, use capital letters as you did for other SQL statements.

**Review by vladsha-dev — CHANGES_REQUESTED** [review]

File: src/django/api/reports/history_of_contributor_uploads.sql (hunk 2 above)
Comment (Innavin369) [reply src/django/api/reports/history_of_contributor_uploads.sql]: Done

File: doc/release/RELEASE-NOTES.md (hunk 1 above)
Comment (Innavin369) [reply doc/release/RELEASE-NOTES.md]: Done

**Review by roman-stolar — DISMISSED** [review]
LGTM

File: doc/release/RELEASE-NOTES.md [hunk 3]
```diff
@@ -28,6 +28,11 @@ This project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html
 * [OSDEV-705](https://opensupplyhub.atlassian.net/browse/OSDEV-705) - Fixed the error “could not convert string to float” that occurred when a list contained columns for “lat” and “lng” and only some of the rows in these columns had data. As a result, rows are processed regardless of whether the values for “lat” and “lng” are present and valid, invalid, or empty.
 
 ### What's new
+* [OSDEV-981](https://opensupplyhub.atlassian.net/browse/OSDEV-981) Reporting. History of contributor uploads. Created a new report with details about the contributor:
+* including name, ID, contributor type;
+* first upload, including date of the first upload and time since the first upload in days;
+* most recent (or “last”) upload, including date of the last upload and time since the last upload in days;
+* total (or “lifetime”) uploads and a calculation for uploads per year (= lifetime uploads/time since the first upload (in years)). This data is ordered based on the “date of last upload” column so that contributors who have recently contributed data are at the top of the report.
```
Comment (vladsha-dev) [inline doc/release/RELEASE-NOTES.md:35]: Wrong formatting
Screenshot 2024-07-05 at 12 52 45

**Review by vladsha-dev — CHANGES_REQUESTED** [review]

File: doc/release/RELEASE-NOTES.md (hunk 3 above)
Comment (vladsha-dev) [reply doc/release/RELEASE-NOTES.md:35]: Please refer to [the docs](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#nested-lists) on how to create a nested list in the .md file. They use hyphens, but you can use asterisks as well.

Comment (Innavin369) [reply doc/release/RELEASE-NOTES.md]: replaced with dash

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Review by roman-stolar — APPROVED** [review]
LGTM

**Pull request merged**